src/graph_coalescence/graph_coalescer.py

```diff
@@ -117,36 +115,43 @@ def coalesce_by_graph(opportunities, predicates_to_exclude=None, coalesce_thresh
                 threshold = len(nodes)
             if i >= threshold:
                 break
-            # for the first 78 enriched result,
+
             # Extract the pvalue and the set of chemical nodes that mapped the enriched link tuples
             best_enrich_p = link[0]
             best_grouping = link[7]
-            # I don't think this is right
-            # best_enrichments = list(filter(lambda x: (x[0] == best_enrich_p) and x[7] == best_grouping,enriched_links))
-            best_enrichments = [link]  # ?
-            # print(best_enrichments)
+            # Port forwarding generates: 'biolink:ameliorates'
+            # Local redis generates: {'predicate': 'biolink:ameliorates'}
+            if (isinstance(link[2], str) or isinstance(link[2], bytes)) and 'predicate' in link[2]:
+                best_enrich_predicate_dict = ast.literal_eval(link[2])
+                best_enrich_predicate = best_enrich_predicate_dict["predicate"]
+            else:
+                best_enrich_predicate = link[2]
+
+            best_enrich_predicate
+            best_enrichments = [link]  #
             attributes = []
 
-            attributes.append({'original_attribute_name': 'coalescence_method',
-                               'attribute_type_id': 'biolink:has_attribute',
+            attributes.append({'attribute_type_id': 'biolink:has_attribute',
                                'value': 'graph_enrichment',
-                               'value_type_id': 'EDAM:operation_0004'})
+                               'value_type_id': 'EDAM:operation_0004',
+                               'original_attribute_name': 'coalescence_method'})
 
-            attributes.append({'original_attribute_name': 'p_value',
-                               'attribute_type_id': 'biolink:has_numeric_value',
+            attributes.append({'attribute_type_id': 'biolink:has_numeric_value',
                                'value': best_enrich_p,
-                               'value_type_id': 'EDAM:data_1669'})
+                               'value_type_id': 'EDAM:data_1669',
+                               'original_attribute_name': 'p_value'})
 
-            attributes.append({'original_attribute_name': 'enriched_nodes',
-                               'attribute_type_id': 'biolink:has_attribute',
+            attributes.append({'attribute_type_id': 'biolink:has_attribute',
                                'value': [x[1] for x in best_enrichments],
-                               'value_type_id': 'EDAM:data_0006'})
+                               'value_type_id': 'EDAM:data_0006',
+                               'original_attribute_name': 'enriched_nodes'})
 
-            newprops = {'attributes': attributes}
+            attributes.append({'attribute_type_id': 'biolink:has_attribute',
+                               'value': [best_enrich_predicate],
```

Comment:
Why a list?  It looks like a single value?

Reply:
Only did that because of the list format in:
`
attributes.append({'attribute_type_id': 'biolink:has_attribute',  'value': [x[1] for x in best_enrichments],`